api/src/application/user/user-service.js

```diff
@@ -30,49 +31,73 @@ module.exports = (bookshelf, emailService, renderTemplate) => {
       return model.fetchAll();
     },
 
-    sendInvite(user, appName, hoursTillExpiration) {
+    sendInvite(user, appName, clientId, redirect_uri, scope, hoursTillExpiration) {
       return self.createPasswordResetToken(user.get('id'), hoursTillExpiration).then(token => {
         const base = config('/baseUrl');
+        const url = encodeURI(`${base}/user/accept-invite?token=${token.get('token')}&client_id=${clientId}&redirect_uri=${redirect_uri}&scope=${scope}`);
         return renderTemplate('email/invite', {
-          url: `${base}/user/accept-invite?token=${token.get('token')}`,
+          url: url.replace(' ', '%20'),
           appName: appName
-        });
-      }).then(emailBody => {
-        return emailService.send({
-          to: user.get('email'),
-          subject: `${appName} Invitation`,
-          html: emailBody,
-        });
+        }).then(emailBody => {
+          return emailService.send({
+            to: user.get('email'),
+            subject: `${appName} Invitation`,
+            html: emailBody,
+          }).then(data => {
+            return data;
+          }).catch(error => {
+            Boom.badImplementation('Something went wrong', error);
+          });
+        }).catch(error => {
+          Boom.badImplementation('Something went wrong', error);
+        });  
+      }).catch(error => {
+        Boom.badImplementation('Something went wrong', error);
       });
     },
 
-    resendUserInvite(userId, appName, hoursTillExpiration) {
+    resendUserInvite(userId, appName, clientId, redirect_uri, scope, hoursTillExpiration) {
       return bookshelf.model('user').where({ id: userId }).fetch().then(user => {
         if (!user) {
           return Boom.notFound();
         }
+        return clientService.findByRedirectUriAndClientId(payload.client_id, payload.redirect_uri).then(clients => {
+          if (clients.models.length === 0) {
+            return Boom.badData('The provided redirect uri is invalid for the given client id.');
```

Comment:
Instead of returning the error object `throw` it.

`throw Boom.badData('The provided redirect uri is invalid for the given client id.');`

throwing from inside of a `.then()` call is the same thing as rejecting the promise.